### PR TITLE
Firefly-1456: generalize and clean up DLGeneratedDropDown.js

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1456](https://jira.ipac.caltech.edu/browse/FIREFLY-1456): generalize and clean up DLGeneratedDropDown.js
  - generalize for multiple projects
  - clean up
  - refactored parts into 3 files
  - fixed upload panel issues when loading a datalink file directly into app
  - fixed the url api for generic firefly app
  - cleaned up TargetHipsPanel to use an embeded primary toolbar

#### Testing
  -  https://firefly-1456-dl-panel.irsakudev.ipac.caltech.edu/irsaviewer/dce
  - use DCE
  - only obvious UI change - DCE hips toolbar

### CODE REVIEW

File: src/firefly/js/ui/DropDownContainer.jsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const dropDownMap = {
     MultiTableSearchCmd: {view: <MultiSearchPanel/>,  layout: {width: '100%'}},
     FileUploadDropDownCmd: {view: <FileUploadDropdown />, layout: {width: '100%'}},
     WorkspaceDropDownCmd: {view: <WorkspaceDropdown />},
-    DLGeneratedDropDownCmd: {view: <DLGeneratedDropDown/>, layout: {width: '100%'}},
+    DLGeneratedDropDownCmd: {view: <DLGeneratedDropDown key='DLGeneratedDropDownCmd' loadRegistry={false}/>, layout: {width: '100%'}},
```
Comment (loitly) [inline src/firefly/js/ui/DropDownContainer.jsx:45]: Why do you need a key here, especially when it's a constant?  If this is not a react's `key`, then maybe name it something else.

File: src/firefly/js/visualize/iv/ImageViewer.jsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ function getStoreState(plotId, oldState) {
 
 const TEN_SECONDS= 10000;
 
-export const ImageViewer= memo( ({showWhenExpanded=false, plotId}) => {
+export const ImageViewer= memo( ({showWhenExpanded=false, plotId, makeToolbar}) => {
```
Comment (loitly) [inline src/firefly/js/visualize/iv/ImageViewer.jsx:64]: Normally, this is a good reason to create slots.  `makeToolbar` would be slotProps.component.

File: src/firefly/js/visualize/reducer/PlotView.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ import {updateTransform, makeTransform} from '../PlotTransformUtils.js';
  * @prop {boolean} [rotateNorthLock]
  * @prop {boolean} [flipYLock]
  * @prop {boolean} [useSticky]
+ * @prop {boolean} [embedMainToolbar] default to false - if true then create the main toolbar within the plot
  * @prop {boolean} [highlightFeedback]
  * @prop {boolean} [useForCoverage=false]
```
Comment (loitly) [inline src/firefly/js/visualize/reducer/PlotView.js:92]: You are already using `useXXX` as switches.  Why not follow suit and call it `useMainToolBar`.

**Review by loitly — APPROVED** [review]
Not exactly sure what to test for since I am not familiar with DCE.  I tried uploading a links table and got a blank `Collection Search` screen.  No error in console.  tableSearch returned a valid tableModel.

File: src/firefly/js/ui/DropDownContainer.jsx (hunk 1 above)
Comment (robyww) [reply src/firefly/js/ui/DropDownContainer.jsx:45]: key make it reset. If there is more than one the useState() will not reset.

File: src/firefly/js/visualize/iv/ImageViewer.jsx (hunk 2 above)
Comment (robyww) [reply src/firefly/js/visualize/iv/ImageViewer.jsx:64]: yes, I plan to do that in the future.

**Pull request merged**